### PR TITLE
Fixes #1025 : Replace '!=' check for comparing symbols in AbstractSpe… 

### LINKED ISSUE #1025: Inline temp doesn't remove casts in some specific cases in VB
Try inlining `x` in the following code:

```VB.NET
Class C
    Function M(args As String()) As Boolean
        Dim x = args(0)
        Return x?.Length = 0
    End Function
End Class
```

**Expected:**
```VB.NET
Class C
    Function M(args As String()) As Boolean
        Return args(0)?.Length = 0
    End Function
End Class
```

**Actual:**
```VB.NET
Class C
    Function M(args As String()) As Boolean
        Return CType(args(CInt(0)), String)?.Length = 0
    End Function
End Class
```

### PR DESCRIPTION
**Pull request by mavasani**
…culationAnalyzer with SymbolsAreCompatible check, which basically uses Equals method for comparison. This was causing two equivalent symbols across semantic model and it's speculative model to be reported as incompatible, causing all simplifications to bail out for ?. expression in VB.

@srivatsn @shyamnamboodiripad @heejaechang can you please review?

### CODE REVIEW

File: src/EditorFeatures/VisualBasicTest/CodeActions/InlineTemporary/InlineTemporaryTests.vb [hunk 1]
```diff
@@ -3864,7 +3864,8 @@ End Class
             Test(code, expected, compareTokens:=False)
         End Sub
 
-        <Fact(Skip:="https://github.com/dotnet/roslyn/issues/1025"), Trait(Traits.Feature, Traits.Features.CodeActionsInlineTemporary)>
+        <WorkItem(1025)>
```
Comment (tmeschter) [inline src/EditorFeatures/VisualBasicTest/CodeActions/InlineTemporary/InlineTemporaryTests.vb]: Include the GitHub link the in `WorkItem` attribute.

**Comment by tmeschter** [comment]
LGTM, modulo my one comment.

**Comment by srivatsn** [comment]
:+1:

File: src/EditorFeatures/VisualBasicTest/CodeActions/InlineTemporary/InlineTemporaryTests.vb (hunk 1 above)
Comment (mavasani) [inline src/EditorFeatures/VisualBasicTest/CodeActions/InlineTemporary/InlineTemporaryTests.vb]: Fixed in the new iteration.

**Comment by shyamnamboodiripad** [comment]
:+1:

**Pull request merged**